process/dataValidators/txValidator.go

```diff
@@ -35,28 +33,29 @@ func NewTxValidator(
 	}
 
 	return &txValidator{
-		accounts:         accounts,
-		shardCoordinator: shardCoordinator,
-		rejectedTxs:      uint64(0),
+		accounts:             accounts,
+		shardCoordinator:     shardCoordinator,
+		rejectedTxs:          uint64(0),
 		maxNonceDeltaAllowed: maxNonceDeltaAllowed,
 	}, nil
 }
 
-// IsTxValidForProcessing will filter transactions that needs to be added in pools
-func (tv *txValidator) IsTxValidForProcessing(interceptedTx process.TxValidatorHandler) bool {
+// CheckTxValidity will filter transactions that needs to be added in pools
+func (tv *txValidator) CheckTxValidity(interceptedTx process.TxValidatorHandler) error {
 	shardId := tv.shardCoordinator.SelfId()
 	txShardId := interceptedTx.SenderShardId()
 	senderIsInAnotherShard := shardId != txShardId
 	if senderIsInAnotherShard {
-		return true
+		return nil
 	}
 
 	sndAddr := interceptedTx.SenderAddress()
 	accountHandler, err := tv.accounts.GetExistingAccount(sndAddr)
 	if err != nil {
-		log.Debug(fmt.Sprintf("Transaction's sender address %s does not exist in current shard %d", sndAddr, shardId))
-		tv.rejectedTxs++
-		return false
+		sndAddrBytes := sndAddr.Bytes()
+		return errors.New(fmt.Sprintf("Transaction's sender address %s does not exist in current shard %d",
```

Comment:
you could do errors.New("Transaction's sender address " + hex.encodetostring(sndAddrBytes)) + " does not "...

Reply:
I would keep it under `fmt.Sprintf` because it is more extensible

Reply:
not really. extensability it is the same.